pgjdbc/src/main/java/org/postgresql/util/ByteConverter.java

```diff
@@ -35,6 +46,150 @@ public static int bytesToInt(byte []bytes) {
+      short d1 = (short)(dig / pow);
+      dig -= d1 * pow;
+      boolean putit = (d1 > 0);
+      if (putit || alwaysPutIt) {
+        buffer.put((char)(d1 + '0'));
+      }
+    }
+
+    buffer.put((char)(dig + '0'));
+  }
+
+  /**
+   * Convert a number from binary representation to text representation.
+   * @param digits array of shorts that can be decoded as the number String
+   * @param scale the scale of the number binary representation
+   * @param weight the weight of the number binary representation
+   * @param sign the sign of the number
+   * @return String the number as String
+   */
+  private static String numberBytesToString(short[] digits, int scale, int weight, int sign) {
+    CharBuffer buffer;
+    int i;
+    int d;
+
+    /*
+     * Allocate space for the result.
+     *
+     * i is set to the # of decimal digits before decimal point. dscale is the
+     * # of decimal digits we will print after decimal point. We may generate
+     * as many as DEC_DIGITS-1 excess digits at the end, and in addition we
+     * need room for sign, decimal point, null terminator.
+     */
+    i = (weight + 1) * DEC_DIGITS;
+    if (i <= 0) {
+      i = 1;
+    }
+
+    buffer = CharBuffer.allocate((i + scale + DEC_DIGITS + 2));
+
+    /*
+     * Output a dash for negative values
+     */
+    if (sign == NUMERIC_NEG) {
+      buffer.put('-');
+    }
+
+    /*
+     * Output all digits before the decimal point
+     */
+    if (weight < 0) {
+      d = weight + 1;
+      buffer.put('0');
+    } else {
+      for (d = 0; d <= weight; d++) {
+        /* In the first digit, suppress extra leading decimal zeroes */
+        digitToString(d, digits, buffer, d != 0);
+      }
+    }
+
+    /*
+     * If requested, output a decimal point and all the digits that follow it.
+     * We initially put out a multiple of DEC_DIGITS digits, then truncate if
+     * needed.
+     */
+    if (scale > 0) {
+      buffer.put('.');
+      for (i = 0; i < scale; d++, i += DEC_DIGITS) {
+        digitToString(d, digits, buffer, true);
+      }
+    }
+
+    /*
+     * terminate the string and return it
+     */
+    int extra = (i - scale) % DEC_DIGITS;
+    return new String(buffer.array(), 0, buffer.position() - extra);
+  }
+
+  public static Number numeric(byte [] bytes) {
+    return numeric(bytes, 0, bytes.length);
+  }
+
+  /**
+   * Convert a variable length array of bytes to an integer
+   * @param bytes array of bytes that can be decoded as an integer
+   * @return integer
+   */
+  public static Number numeric(byte [] bytes, int pos, int numBytes) {
+    if (numBytes < 8) {
+      throw new IllegalArgumentException("number of bytes should be at-least 8");
+    }
+
+    short len = ByteConverter.int2(bytes, pos);
+    short weight = ByteConverter.int2(bytes, pos + 2);
+    short sign = ByteConverter.int2(bytes, pos + 4);
+    short scale = ByteConverter.int2(bytes, pos + 6);
+
+    if (numBytes != (len * Short.BYTES) + 8) {
```

Comment:
Short.BYTES is actually Short.SIZE/Byte.SIZE and is only available in java 1.8 and up